src/main/java/com/scaler/lld/design/assignments/builder/DatabaseConfigurationBuilder.java

```diff
@@ -1,6 +1,95 @@
 package com.scaler.lld.design.assignments.builder;
 
+import lombok.Getter;
+import lombok.Setter;
+
 @WithBuilder
+
 public class DatabaseConfigurationBuilder {
 
+    private String databaseUrl;
+    private String username;
+    private String password;
+    private int maxConnections;
+    private boolean enableCache;
+    private boolean isReadOnly;
+
+    public String getDatabaseUrl() {
+        return databaseUrl;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public int getMaxConnections() {
+        return maxConnections;
+    }
+
+    public boolean isEnableCache() {
+        return enableCache;
+    }
+
+    public boolean isReadOnly() {
+        return isReadOnly;
+    }
+
+    private DatabaseConfigurationBuilder(String databaseUrl, String username, String password, int maxConnections, boolean enableCache, boolean isReadOnly) {
+            this.databaseUrl = databaseUrl;
+            this.username = username;
+            this.password = password;
+            this.maxConnections = maxConnections;
+            this.enableCache = enableCache;
+            this.isReadOnly = isReadOnly;
+    }
+
+    public static Builder getBuilder(){
+        return new Builder();
+    }
+
+    @Setter
+    @Getter
+    public static class Builder{
+
+        private String databaseUrl;
+        private String username;
+        private String password;
+        private int maxConnections;
+        private boolean enableCache;
+        private boolean isReadOnly;
+
+        public void setDatabaseUrl(String databaseUrl) {
```

Comment:
These setters work, however you improve the consumer's code by implementing method chaining like
```java
builder.setUsername("Tantia").setPassword("NotToday")
```
Try to implement the above.